### ISSUE TITLE
"#2014 - Commands out of sync; you can't run this command now" returned when stored procedure does not return any result set

### ISSUE THREAD

**Issue by javide** [phpmyadmin/phpmyadmin#14614]
### Describe the bug

In **phpMyAdmin 4.8.3** if you call a stored procedure that contains a SELECT statement, when this SELECT does not return a result set, this error is displayed:
`#2014 - Commands out of sync; you can't run this command now`

When calling the same stored procedure from the mysql command line the expected output is returned instead:
```
Empty set (0.70 sec)
Query OK, 0 rows affected (0.70 sec)
```

### Server configuration

- Operating system:
- Web server:
- Database version: Server version: 5.5.5-10.1.19-MariaDB MariaDB Server
- PHP version:
- phpMyAdmin version: phpMyAdmin 4.8.3

### Client configuration

- Browser:
- Operating system:

**Comment by williamdes** [comment]
Stackoverflow: https://stackoverflow.com/a/20435243/5155484

**Comment by williamdes** [comment]
Unable do reproduce using phpMyAdmin 4.8.3 and :
- MariaDB 10.2.16
- MySQL 5.7.19
- MariaDB 10.0.35
- MySQL 5.6.38
- Demo server 4.8.x https://demo.phpmyadmin.net/QA_4_8/server_sql.php

**Comment by williamdes** [comment]
@javide I remember having this issue, can you please explain how to reproduce this issue ?
Maybe can you try on https://demo.phpmyadmin.net/QA_4_8/ ?

**Comment by williamdes** [comment]
Create the procedure
```sql
DELIMITER $$
CREATE PROCEDURE `abc`()
    NO SQL
(SELECT * FROM mysql.user WHERE 0)
UNION
(SELECT * FROM mysql.user WHERE 0)$$
DELIMITER ;
```
And run
```sql
CALL abc()
```

**Comment by williamdes** [comment]
Found the solution on https://community-auth.com/blog-posts/commands-out-of-sync-error-when-using-stored-procedures

Fixed by 42a4fbbcc6c0e58e0abe6be2e2d4c9950b018271
